openlibrary/catalog/add_book/__init__.py

```diff
@@ -522,17 +525,20 @@ def find_exact_match(rec, edition_pool):
     return False
 
 
-def find_match(e1, edition_pool):
+def find_enriched_match(rec, edition_pool):
     """
-    Find the best match for e1 in edition_pool and return its key.
-    :param dict e1: the new edition we are trying to match, output of build_marc(import record)
-    :param list edition_pool: list of possible edition matches, output of build_pool(import record)
+    Find the best match for rec in edition_pool and return its key.
+    :param dict rec: the new edition we are trying to match.
+    :param list edition_pool: list of possible edition key matches, output of build_pool(import record)
     :rtype: str|None
-    :return: None or the edition key '/books/OL...M' of the best edition match for e1 in edition_pool
+    :return: None or the edition key '/books/OL...M' of the best edition match for enriched_rec in edition_pool
     """
+    enriched_rec = build_marc(rec)
+    add_db_name(enriched_rec)
```

Comment:
We might want to rename this to something like `add_author_fingerprint` (where fingerprint is a unique string; I assume this is used to avoid duplicate author etc, using `{name}_{birth}_{death}` as a unique fingerprint for the author).

```suggestion
    add_db_name(enriched_rec)
```

Actually we might want to further simplify this by making that function pure, eg `compute_author_fingerprint(author: dict) -> str` and then do the updating of the records here? If that makes sense eg 
```
rec['authors'] = [
    {**a, 'fingerprint': compute_author_fingerprint(a)}
    for a in authors
]
```

Up to you if you think that would be better in a separate PR.